scala/src/main/scala/com/xebia/functional/xef/scala/auto/package.scala

```diff
@@ -1,101 +1,97 @@
+    chat: ChatWithFunctions = OpenAI.FromEnvironment.DEFAULT_SERIALIZATION,
     promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS
-)(using scope: AIScope): A =
-  LoomAdapter.apply((cont) =>
-    scope.kt.promptWithSerializer[A](
-      llmModel,
-      prompt,
-      generateCFunctions.asJava,
-      (json: String) => parse(json).flatMap(Decoder[A].decodeJson(_)).fold(throw _, identity),
-      promptConfiguration,
-      cont
-    )
-  )
-
-private def generateCFunctions[A: SerialDescriptor]: List[CFunction] =
-  val descriptor = SerialDescriptor[A].serialDescriptor
-  val serialName = descriptor.getSerialName
-  val fnName =
-    if (serialName.contains(".")) serialName.substring(serialName.lastIndexOf("."), serialName.length)
-    else serialName
-  List(CFunction(fnName, "Generated function for $fnName", Json.encodeJsonSchema(descriptor)))
-
-def addContext(docs: Iterable[String])(using scope: AIScope): Unit =
-  LoomAdapter.apply(scope.kt.addContext(docs.asJava, _))
+)(using
+    conversation: ScalaConversation
+): A =
+  val fromJson = new FromJson[A] {
+    def fromJson(json: String): A =
+      parse(json).flatMap(Decoder[A].decodeJson(_)).fold(throw _, identity)
+  }
+  conversation.prompt(chat, prompt, generateCFunctions.asJava, fromJson, promptConfiguration).join()
 
 def promptMessage(
-    prompt: String,
-    llmModel: Chat = OpenAI().DEFAULT_CHAT,
+    question: String,
+    chat: Chat = OpenAI.FromEnvironment.DEFAULT_CHAT,
     promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS
-)(using scope: AIScope): String =
-  LoomAdapter
-    .apply[String](
-      scope.kt.promptMessage(llmModel, prompt, promptConfiguration, _)
-    )
+)(using conversation: ScalaConversation): String =
+  conversation.promptMessage(chat, question, promptConfiguration).join()
 
 def promptMessages(
-    prompt: String,
-    llmModel: Chat = OpenAI().DEFAULT_CHAT,
-    functions: List[CFunction] = List.empty,
+    question: String,
+    chat: Chat = OpenAI.FromEnvironment.DEFAULT_CHAT,
+    functions: List[CFunction] = List(),
     promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS
-)(using scope: AIScope): List[String] =
-  LoomAdapter
-    .apply[java.util.List[String]](
-      scope.kt.promptMessages(llmModel, prompt, functions.asJava, promptConfiguration, _)
-    ).asScala.toList
-
-def pdf(
-    resource: String | File,
-    splitter: TextSplitter = TextSplitter.tokenTextSplitter(ModelType.getDEFAULT_SPLITTER_MODEL, 100, 50)
+)(using
+    conversation: ScalaConversation
 ): List[String] =
-  LoomAdapter
-    .apply[java.util.List[String]](count =>
-      resource match
-        case url: String => Loader.pdf(url, splitter.core, count)
-        case file: File => Loader.pdf(file, splitter.core, count)
-    ).asScala.toList
+  conversation.promptMessages(chat, question, functions.asJava, promptConfiguration).join().asScala.toList
```

Comment:
We assume LOOM for now, and we block here, in practice scala users may want to consume this blocking, with CompletableFuture and with an IO